libs/coin-framework/src/operation.ts

```diff
@@ -154,13 +154,17 @@ export function flattenOperationWithInternalsAndNfts(op: Operation): Operation[]
   return ops;
 }
 
-export function getOperationAmountNumber(op: Operation): BigNumber {
+export function getOperationAmountNumber(
+  op: Operation,
+  fromSubAccount?: boolean,
+  getFeeInstead?: boolean,
+): BigNumber {
   switch (op.type) {
     case "IN":
     case "REWARD":
     case "REWARD_PAYOUT":
     case "WITHDRAW":
-      return op.value;
+      return getFeeInstead ? BigNumber(0) : op.value;
```

Comment:
Logic is becoming too hard to understand and shouldn't be driven by the caller

We can discuss it on slack @semeano , it's related to tests ?